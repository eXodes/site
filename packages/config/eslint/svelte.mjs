import { rules } from "./rules.mjs";
import eslintConfigPrettier from "eslint-config-prettier";
import eslintConfigTurbo from "eslint-config-turbo/flat";
import onlyWarn from "eslint-plugin-only-warn";
import eslintPluginSvelte from "eslint-plugin-svelte";
import eslintPluginUnicorn from "eslint-plugin-unicorn";
import globals from "globals";
import svelteParser from "svelte-eslint-parser";
import tsEslint from "typescript-eslint";

export default tsEslint.config(
  eslintPluginUnicorn.configs["flat/recommended"],
  ...tsEslint.configs.recommended,
  ...tsEslint.configs.stylistic,
  ...eslintPluginSvelte.configs["flat/recommended"],
  eslintConfigPrettier,
  ...eslintPluginSvelte.configs["flat/prettier"],
  ...eslintConfigTurbo,
  {
    plugins: {
      "only-warn": onlyWarn,
    },

    languageOptions: {
      globals: { ...globals.browser, ...globals.node },

      ecmaVersion: "latest",
      sourceType: "module",

      parser: tsEslint.parser,
      parserOptions: {
        tsconfigRootDir: import.meta.dirname,
        parser: tsEslint.parser,
      },
    },

    rules: {
      ...rules,
    },
  },
  {
    files: ["**/*.svelte"],

    languageOptions: {
      ecmaVersion: 5,
      sourceType: "script",

      parser: svelteParser,
      parserOptions: {
        parser: tsEslint.parser,
        extraFileExtensions: [".svelte"],
      },
    },
  }
);
