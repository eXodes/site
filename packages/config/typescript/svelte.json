{
  "display": "Svelte",
  "compilerOptions": {
    "allowJs": true,
    "checkJs": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "isolatedModules": true,
    "lib": ["esnext", "dom", "dom.iterable"],
    "module": "esnext",
    "moduleDetection": "force",
    "moduleResolution": "bundler",
    "noEmit": true,
    "noUncheckedIndexedAccess": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "sourceMap": true,
    "strict": true,
    "target": "esnext"
  }
}
