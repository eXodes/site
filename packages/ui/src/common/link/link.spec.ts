import Link from "./link.svelte";
import { render } from "@testing-library/svelte";
import { createRawSnippet } from "svelte";
import { describe, expect, test } from "vitest";

describe("Link", () => {
  test("should render component", async () => {
    const { container } = render(Link, {
      href: "/login",
      children: createRawSnippet(() => {
        return {
          render: () => `<span>Login</span>`,
        };
      }),
    });

    expect(container.children).toMatchSnapshot();
  });

  test("should render with external", async () => {
    const { container } = render(Link, {
      href: "https://acme.inc",
      external: true,
      children: createRawSnippet(() => {
        return {
          render: () => `<span>Login</span>`,
        };
      }),
    });

    expect(container.children).toMatchSnapshot();
  });
});
