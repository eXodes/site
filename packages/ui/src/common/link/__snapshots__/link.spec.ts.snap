// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link > should render component 1`] = `
HTMLCollection [
  <a
    class="-mx-0.5 border-b-4 border-yellow bg-gradient-to-b from-transparent from-50% to-yellow to-50% bg-[length:100%_200%] px-0.5 font-medium transition-all duration-500 hover:border-b-2 hover:bg-[0_100%]"
    href="/login"
  >
    <span>
      Login
    </span>
    <!---->
  </a>,
]
`;

exports[`Link > should render with external 1`] = `
HTMLCollection [
  <a
    class="-mx-0.5 border-b-4 border-yellow bg-gradient-to-b from-transparent from-50% to-yellow to-50% bg-[length:100%_200%] px-0.5 font-medium transition-all duration-500 hover:border-b-2 hover:bg-[0_100%]"
    href="https://acme.inc"
    target="_blank"
  >
    <span>
      Login
    </span>
    <!---->
  </a>,
]
`;
